### PR TITLE
Feature/events

### PR DESCRIPTION
**Pull request by JbirdL86**
What's new :question: 

- Added calculate.js and operate.js under the logic folder
- Added constructor to Calculator component
- Added binding method to its constructor

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,44 +1,60 @@
...
+          <button type="button" onClick={this.onClickHandler} className="number">2</button>
+          <button type="button" onClick={this.onClickHandler} className="number">3</button>
+          <button type="button" onClick={this.onClickHandler} className="operation right">+</button>
         </div>
         <div className="row bottom">
-          <div className="number zero">0</div>
-          <div className="operation">.</div>
-          <div className="operation right">=</div>
+          <button type="button" onClick={this.onClickHandler} className="number zero">0</button>
+          <button type="button" onClick={this.onClickHandler} className="operation">.</button>
```
Comment (oxenprogrammer) [inline src/components/Calculator.js]: **Changes Required**
- [ ] There is too much repetition of the `button` here making our code not follow the `Don't Repeat Yourself` principle. Let us try to refactor this into a `Button Component` and  pass the name and onClick event as `props`.
- [ ] Let us try to store all the button names in an `array`. In that way, we can then use `array.map` to dynamically add all the buttons into the `Virtual DOM`.

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,44 +1,60 @@
...
+    this.state = {
+      total: null,
+      next: null,
+    };
+    this.onClickHandler = this.onClickHandler.bind(this);
+  }
+
+  onClickHandler(e) {
+    this.setState((state) => calculate(state, e.target.innerHTML));
+  }
```
Comment (oxenprogrammer) [inline src/components/Calculator.js:18]: **Changes Required**
- [ ] Every time a button is clicked, the state update will re-render the `virtual DOM`. To handle this in `Class Components`, we need to usually wrap this `side-effect` inside lifecycle methods like `ComponentDidMount` and `ComponentWillUpdate`. Let us try to use lifecycle methods to handle this DOM side effect.

**Review by oxenprogrammer — CHANGES_REQUESTED** [review]
Hi @JbirdL86 , 

Good job so far!
**To Highlight**
- [x] Professional Repo.
- [x] The Calculator works perfectly. 
I can see you're really putting effort to become a world-class software developer :+1:
There are some issues that you still need to address in order to move forward to the next project.
Trust me you got this! :muscle:


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Consider the optional changes nice to have features and only work on them when you have satisfied the required changes_

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by DeliceLydia — APPROVED** [review]
# Congratulations! 🎉

## Hi @JbirdL86, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**